### PR TITLE
M2P-466 Bugfix: Unable to remove items from Cart with Update Cart Endpoint

### PR DESCRIPTION
**Pull request by dumega**
# Description
If the customer tries to remove an item from the cart, there is no need to check its valid quantity in the store.

Fixes: https://boltpay.atlassian.net/browse/M2P-466

#changelog Bugfix: Unable to remove items from Cart with Update Cart Endpoint

# Type of change

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

**Review by vitaliyreznikov — APPROVED** [review]
Would you mind to add test for this case? Maybe even in separate PR, because I am going to create RC tag right now.

File: Model/Api/UpdateCartItemTrait.php [hunk 1]
```diff
@@ -130,6 +130,10 @@ protected function verifyItemData($product, $updateItem, $quoteItem, $websiteId)
             $qtyToCheck = ($updateItem['update'] == 'add')
                     ? (int)$quoteItem['quantity'] + (int)$updateItem['quantity']
                     : (int)$quoteItem['quantity'] - (int)$updateItem['quantity'];
+            
+            if ($updateItem['update'] == 'remove' && $qtyToCheck == 0) {
```
Comment (danac-gs) [inline Model/Api/UpdateCartItemTrait.php:134]: Why do we need `$qtyToCheck == 0` check here?

Comment (dumega) [reply Model/Api/UpdateCartItemTrait.php:134]: if the customer want to remove the item from cart, the update-cart api hook would have the quantity value of request equal to quote item quantity, so $qtyToCheck == 0 means the same behavior. 
Then $qtyToCheck > 0 that's for updating quantity
$qtyToCheck < 0 I believe that's an exception,

**Comment by dumega** [comment]
@vitaliyreznikov We already have unit test for this case. https://github.com/BoltApp/bolt-magento2/blob/2.18.1/Test/Unit/Model/Api/UpdateCartItemTraitTest.php#L384

File: Model/Api/UpdateCartItemTrait.php (hunk 1 above)
Comment (danac-gs) [reply Model/Api/UpdateCartItemTrait.php:134]: gotcha. if we have, say, 2 units of some item in cart, we can currently only remove both (all) of them via the update cart request. not decreasing the quantity to 1. correct?

Comment (dumega) [reply Model/Api/UpdateCartItemTrait.php:134]: @danac-gs 
1. Under current logic, the update-cart does not allow empty the cart, so if you try to remove the last item, the Bolt modal would directly block that behavior.
2. We can decrease the quantity to 1.
3. If you set the quantity of item to 0 on the Bolt modal, it actually remove the item.

Comment (vitaliyreznikov) [reply Model/Api/UpdateCartItemTrait.php:134]: @dumega Question here: if we have 3 unit of the same item in cart and going to decrease it to 2. Should we check inventory in this case or not? Going to merge this PR, we can discuss this case separately.

**Review by vitaliyreznikov — APPROVED** [review]

File: Model/Api/UpdateCartItemTrait.php (hunk 1 above)
Comment (dumega) [reply Model/Api/UpdateCartItemTrait.php:134]: @vitaliyreznikov we only ignore the quantity checking if $qtyToCheck == 0, cause that means the customer try to remove the item from cart, then for other cases, we would always process validation.

**Pull request merged**